### ISSUE TITLE
Add mock examples to apollo-link-schema README

### ISSUE THREAD

**Issue by anthonator** [apollographql/apollo-link#535]
It would be good to have some examples of mocking for `apollo-link-schema`. Currently, the mock examples aren't very helpful:

```js
const mocks = {
  Query: () => ...,
  Mutation: () => ...
};
```

It would also be good to have some different mock scenarios. For example, mocks that demonstrate:

* successful requests
* requests that return `graphQLErrors`
* requests that return `networkErrors`

I'm assuming that mocks are expected to be functions that either return a payload or throw an `ApolloError`. Others may need some more guidance.

**Comment by evans** [comment]
I totally agree that would be great! I added some more documentation around mocking. #363 adds a little and points at the graphql-tools mocking functions. Could you take a look and see what should be added there?

/label docs
